Gridy/Controller/PuzzleViewController.swift

```diff
@@ -35,22 +57,115 @@ class PuzzleViewController: UIViewController, UIGestureRecognizerDelegate {
         self.newGameButton.clipsToBounds = true
     }
     
-    @IBAction func startNewGame(_ sender: Any) {
-        dismiss(animated: true)
-    }
-    
     private func setupPuzzlePiecesImageViews() {
         let randomizedImagePieces = imagePieces.shuffled()
         
         for (index, puzzlePiece) in puzzlePiecesImageViews.enumerated() {
-            //setup image views borders
-            puzzlePiece.layer.borderColor = UIColor(named: Constant.Color.primaryLight)?.cgColor
-            puzzlePiece.layer.masksToBounds = true
-            puzzlePiece.contentMode = .scaleToFill
-            puzzlePiece.layer.borderWidth = 1
-            //fill puzzle pieces with images
+            puzzlePiece.translatesAutoresizingMaskIntoConstraints = true
             puzzlePiece.image = randomizedImagePieces[index]
         }
     }
     
+    private func updatePuzzlePiecesImageViews() {
+        for (index, puzzlePiece) in puzzlePiecesImageViews.enumerated() {
+            if (puzzlePiece.tag >= 1 && puzzlePiece.tag <= 16) {
+                puzzlePiece.frame = self.view.convert(puzzleBlocksViews[puzzlePiece.tag - 1].bounds, from: puzzleBlocksViews[puzzlePiece.tag - 1])
+            } else if (puzzlePiece.tag >= 17 && puzzlePiece.tag <= 33) {
+                puzzlePiece.frame = self.view.convert(puzzlePiecesPlaceholdersViews[puzzlePiece.tag - 17].bounds, from: puzzlePiecesPlaceholdersViews[puzzlePiece.tag - 17])
+            } else {
+                puzzlePiece.frame = self.view.convert(puzzlePiecesPlaceholdersViews[index].bounds, from: puzzlePiecesPlaceholdersViews[index])
+            }
+        }
+    }
+    
+    private func setupPuzzlePiecesPlaceholdersViews() {
+        for puzzlePiecePlaceholder in puzzlePiecesPlaceholdersViews {
+            //setup image views borders
```

Comment:
The borders could be the reason why you see lines, because borders are not part of the `UIView.frame`